app/src/main/java/com/github/sdp_begreen/begreen/firebase/DB.kt

```diff
@@ -0,0 +1,140 @@
+
+    /**
+     * Adds and [image] for the user [userId] in the database
+     *
+     * @param image the image we want to add to the database
+     * @param userId the ID of the user wanting to store a new image
+     * @return a unique ID under which the image got stored, and null if the image couldn't get stored
+     */
+    suspend fun addImage(image: Bitmap, userId: Int, metadata: PhotoMetadata): PhotoMetadata?
+
+    /**
+     * Test whether a [User] exists in the database for the given [userId]
+     *
+     * @param userId The id of the user to check for the existence in the database
+     * @param timeout the maximum time we wait for the database to respond
+     * @return true if the user exists, false otherwise
+     */
+    suspend fun userExists(userId: String, timeout: Long = TIMEOUT): Boolean
+
+    /**
+     * Retrieves the image associated with the given [userId] and [metadata] from the database
+     *
+     * @param metadata the metadata associated with the given image we want to retrieve
+     * @param userId the ID of the user where we should find the image
+     * @param timeout the maximum time we wait for the database to respond
+     * @return the image, or null if no image was found
+     *
+     * @throws StorageException if the image could not be retrieved
+     * @throws DatabaseTimeoutException if the database could not be reached
+     * @throws DatabaseException if the an exception occurred while retrieving the image
+     */
+    suspend fun getImage(metadata: PhotoMetadata, userId: Int, timeout: Long = TIMEOUT): Bitmap?
+
+    /**
+     * Retrieves the profile image associated with the given [userId] and [metadata] from the database
+     *
+     * @param metadata the metadata associated with the given image we want to retrieve
+     * @param userId the ID of the user where we should find the image
+     * @param timeout the maximum time we wait for the database to respond
+     * @return the image, or null if no image was found
+     * @throws StorageException if the image could not be retrieved
+     * @throws DatabaseTimeoutException if the database could not be reached
+     * @throws DatabaseException if the an exception occurred while retrieving the image
+     */
+    suspend fun getUserProfilePicture(metadata: PhotoMetadata, userId: String, timeout: Long = TIMEOUT): Bitmap?
+
+    /**
+     * Store the given [location] as the location of a recycling bin.
+     *
+     * @param location the location of the bin
+     *
+     * @return true if the location got stored in the database, and false if it failed
+     */
+    suspend fun storeBinLocation(location: LatLng): Boolean
+
+    /**
+     * Retrieves the set of all bin locations currently present in the database
+     *
+     * @return the set of locations of all the bins, or null if an error happened
+     */
+    suspend fun getAllBinLocations(): Set<LatLng>?
+
+    /**
+     * Retrieves the list of advices from the realtime database
```

Comment:
terminology: in English, "advice" is uncountable, so you have "a piece of advice" but not "an advice". Perhaps "tips" is a better word for this.